test/Node_test.lua

```diff
@@ -0,0 +1,56 @@
+luaunit = require("luaunit")
+Node = require("Node")
+
+local TestNode = {}
+
+local root, node_1, node_2, node_3 = Node:new(), Node:new(), Node:new(), Node:new()
+root:add_child(node_1)
+root:add_child(node_2)
+node_2:add_child(node_3)
+
+function TestNode:test_new()
+  luaunit.assertIs(getmetatable(root), Node)
+
+  luaunit.assertNotNil(root)
+
+  luaunit.assertNil(root.parent)
+end
+
+function TestNode:test_add()
```

Comment:
`test_add` hasn't this test already... (each test case needs all assertations in order have a full coverage even if there are overlaps between tests)